Builds fail in pnpm projects using @loopback/rest 1.0.1

## Description / Steps to reproduce / Feature proposal

Trying to use loopback 4 1.0.0/1 in a `pnpm` project.

## Current Behavior

Project fails to build due to `@loopback/rest` package relying on `npm` providing transitive dependencies to it, specifically for `serve-static` and `express-serve-static-core`

## Expected Behavior

`@loopback/` packages should declare all their dependencies so that they work with stricter package managers like `pnpm`

Example error output:
```
> lb-tsc es2018

../../common/temp/node_modules/.registry.npmjs.org/@loopback/rest/1.0.1/node_modules/@loopback/rest/dist/src/rest.application.d.ts:8:36 - error TS2307: Cannot find module 'serve-static'.

8 import { ServeStaticOptions } from 'serve-static';
                                     ~~~~~~~~~~~~~~

../../common/temp/node_modules/.registry.npmjs.org/@loopback/rest/1.0.1/node_modules/@loopback/rest/dist/src/rest.application.d.ts:9:28 - error TS2307: Cannot find module 'express-serve-static-core'.

9 import { PathParams } from 'express-serve-static-core';
                             ~~~~~~~~~~~~~~~~~~~~~~~~~~~

../../common/temp/node_modules/.registry.npmjs.org/@loopback/rest/1.0.1/node_modules/@loopback/rest/dist/src/rest.server.d.ts:8:28 - error TS2307: Cannot find module 'express-serve-static-core'.

8 import { PathParams } from 'express-serve-static-core';
                             ~~~~~~~~~~~~~~~~~~~~~~~~~~~

../../common/temp/node_modules/.registry.npmjs.org/@loopback/rest/1.0.1/node_modules/@loopback/rest/dist/src/rest.server.d.ts:10:36 - error TS2307: Cannot find module 'serve-static'.

10 import { ServeStaticOptions } from 'serve-static';
                                      ~~~~~~~~~~~~~~
```

The `../../common/temp/...` is because this is also a project in a rush monorepo

## Workaround

Add packages to my project's dependencies:
* `serve-static`
* ~`express-serve-static-core`~
  * Actually it turns out it's very important I do _not_ add this as a dependency of my project, because this package only exports types ... wrong types! 
* `@types/serve-static`
* `@types/express-serve-static-core`
  * This one is perversely important, as otherwise the types built into `express-serve-static-core` (which is the sole content of that module) get used, which are _wrong_

--- a/packages/rest/package.json
+++ b/packages/rest/package.json
@@ -43,6 +43,7 @@
     "parseurl": "^1.3.2",
     "path-to-regexp": "^2.2.0",
     "qs": "^6.5.2",
+    "serve-static": "^1.13.2",
     "strong-error-handler": "^3.2.0",
     "validator": "^10.4.0"
   },
@@ -54,7 +55,9 @@
     "@types/debug": "0.0.30",
     "@types/js-yaml": "^3.11.1",
     "@types/lodash": "^4.14.106",
-    "@types/node": "^10.11.2"
+    "@types/node": "^10.11.2",
+    "@types/express-serve-static-core": "^4.16.0",
+    "@types/serve-static": "1.13.2"
   },
   "files": [
     "README.md",

